File: Java/src/main/java/QuestionBank/Solution1401_1500/Solution1446.java
package QuestionBank.Solution1401_1500;
/**
 * @author windows
 */
public class Solution1446 {
    public int maxPower(String s) {
        int count = 0;
        int max = 0;
        for (int i = 1; i < s.length(); i++) {
            if (s.charAt(i) == s.charAt(i - 1)){
                count++;
                max = Math.max(max, count);
            }else {
                count = 1;
            }
        }
        return max;
    }
    
    
    public static void main(String[] args) {
        Solution1446 solution1446 = new Solution1446();
        System.out.println(solution1446.maxPower("leetcode"));
    }
}

File: C++/src/Solution1401_1500/Solution1446.cpp
//
// Created by windows on 2023/3/2.
//

#include "Solution1446.h"
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;

int maxPower(string s) {
  int maxVal = 1;
  int n = s.size();
  for (int i = 0; i < n - 1; ++i) {
    int count = 1;
    for (int j = i + 1; j < n; ++j) {
      if (s[j] == s[j - 1]){
        count++;
      }else{
        break;
      }
    }
    maxVal = max(maxVal, count);
  }
  return maxVal;
}